src/handshakes/abstract-handshake.ts

```diff
@@ -0,0 +1,171 @@
+import {Buffer} from "buffer";
+import { AEAD, x25519, HKDF, SHA256 } from 'bcrypto';
+
+import {bytes, bytes32, uint32} from "../@types/basic";
+import {CipherState, MessageBuffer, SymmetricState} from "../@types/handshake";
+import {getHkdf} from "../utils";
+
+export abstract class AbstractHandshake {
```

Comment:
Those all methods are pure, meaning they don't need to be in class, they can be in file called handshake-utils.

Reply:
This is true but I am not sure if it's the right approach. Made more sense like this to me

Reply:
I don't have any strong opinion on this so...

Reply:
You ain't writing unit tests for handshake so it doesn't have much impact. (It's much more "testable" code if methods are not wrapped inside class). It could have impact if third handshake uses different stuff

Reply:
Got your point. Leaving for now as it's not the case